### PR TITLE
[RFC] Fix build failure using newer versions of msgpack-c

### LINKED ISSUE #1899: Build failure ‘msgpack_object_union’ has no member named ‘dec’
I can't get the current source to build. Doing this on Gentoo with the ebuild from the lua overlay and GCC 4.9.2

```
<built-in>: note: this is the location of the previous definition
/var/tmp/portage/app-editors/neovim-9999/work/neovim-9999/src/nvim/msgpack_rpc/helpers.c: In function ‘msgpack_rpc_to_float’:
/var/tmp/portage/app-editors/neovim-9999/work/neovim-9999/src/nvim/msgpack_rpc/helpers.c:89:18: error: ‘msgpack_object_union’ has no member named ‘dec’
   *arg = obj->via.dec;
                  ^
/var/tmp/portage/app-editors/neovim-9999/work/neovim-9999/src/nvim/msgpack_rpc/helpers.c:90:23: error: ‘MSGPACK_OBJECT_DOUBLE’ undeclared (first use in this function)
   return obj->type == MSGPACK_OBJECT_DOUBLE;
                       ^
/var/tmp/portage/app-editors/neovim-9999/work/neovim-9999/src/nvim/msgpack_rpc/helpers.c:90:23: note: each undeclared identifier is reported only once for each function it appears in
/var/tmp/portage/app-editors/neovim-9999/work/neovim-9999/src/nvim/msgpack_rpc/helpers.c: In function ‘msgpack_rpc_to_object’:
/var/tmp/portage/app-editors/neovim-9999/work/neovim-9999/src/nvim/msgpack_rpc/helpers.c:123:10: error: ‘MSGPACK_OBJECT_DOUBLE’ undeclared (first use in this function)
     case MSGPACK_OBJECT_DOUBLE:
          ^
/var/tmp/portage/app-editors/neovim-9999/work/neovim-9999/src/nvim/msgpack_rpc/helpers.c: In function ‘msgpack_rpc_to_float’:
/var/tmp/portage/app-editors/neovim-9999/work/neovim-9999/src/nvim/msgpack_rpc/helpers.c:91:1: warning: control reaches end of non-void function [-Wreturn-type]
 }
 ^
src/nvim/CMakeFiles/nvim.dir/build.make:2795: recipe for target 'src/nvim/CMakeFiles/nvim.dir/msgpack_rpc/helpers.c.o' failed
make[2]: *** [src/nvim/CMakeFiles/nvim.dir/msgpack_rpc/helpers.c.o] Error 1
make[2]: *** Waiting for unfinished jobs....
/var/tmp/portage/app-editors/neovim-9999/work/neovim-9999/src/nvim/msgpack_rpc/channel.c: In function ‘job_err’:
/var/tmp/portage/app-editors/neovim-9999/work/neovim-9999/src/nvim/msgpack_rpc/channel.c:346:12: warning: unused variable ‘channel’ [-Wunused-variable]
   Channel *channel = job_data(data);
            ^
```

### PR DESCRIPTION
**Pull request by ghost**
Here's the relevant issue and commit in the `msgpack-c` issue tracker:
https://github.com/msgpack/msgpack-c/issues/191
https://github.com/msgpack/msgpack-c/commit/737e6703df2cad06cfb16e3e0e34a02ec5212c38

### CODE REVIEW

**Comment by Pyrohh** [comment]
Repushed with a newer version of msgpack-c, which fixes https://github.com/msgpack/msgpack-c/issues/209.

**Comment by justinmk** [comment]
restarted OS X build

File: .travis.yml [hunk 1]
```diff
@@ -5,7 +5,7 @@ env:
   global:
     - CI_SCRIPTS=$TRAVIS_BUILD_DIR/.ci
     # To build third-party dependencies, set this to 'true'.
-    - BUILD_NVIM_DEPS=false
+    - BUILD_NVIM_DEPS=true
```
Comment (justinmk) [inline .travis.yml]: this is temporary, right?

**Comment by justinmk** [comment]
LGTM. @tarruda could you peek at this?

File: .travis.yml (hunk 1 above)
Comment (Pyrohh) [inline .travis.yml]: yes

**Comment by justinmk** [comment]
@Pyrohh Did you test this with the python client? E.g., just verify that a python plugin like YCM or ultisnips works as usual.

**Comment by Pyrohh** [comment]
@justinmk Just tried out UltiSnips and it works fine using the latest python client.

**Comment by justinmk** [comment]
Just looked around, I didn't see any commits since ecf4b09acd29746829b6a02939db91dfdec635b4 that would be a problem for us.

**Comment by Pyrohh** [comment]
Ooh! Great to see Coveralls working.

**Comment by justinmk** [comment]
Coveralls really needs a "threshold" or epsilon value, to avoid noise.

@Pyrohh If you can send a similar PR to https://github.com/neovim/homebrew-neovim , and remove the temp commit here, I see no reason not to merge this...

**Comment by Pyrohh** [comment]
Done, gonna do the homebrew one now.

**Comment by Pyrohh** [comment]
done

**Comment by justinmk** [comment]
here we go...

**Comment by justinmk** [comment]
merged. Going to run bot-ci now.

**Comment by jszakmeister** [comment]
FYI: this update broke FreeBSD

```text
00:12:08,664 INFO  - /usr/home/quickbuild/buildagent/workspace/root/neovim/master/.deps/build/src/msgpack/include/msgpack/sysdep.h:80:14: fatal error: 'byteswap.h' file not found
00:12:08,664 INFO  - # include <byteswap.h>
00:12:08,664 INFO  - ^
```

**Comment by jszakmeister** [comment]
I added a comment to msgpack/msgpack-c#221.  Hopefully it will get fixed soonish.

**Comment by justinmk** [comment]
@jszakmeister great to see quick feedback from your bsd builds!

**Comment by jszakmeister** [comment]
@justinmk I'm hoping to get my QuickBuild instance in a position that it can be reached by others and have it start checking pull requests too.  I still have a few issues to sort out though.

**Pull request closed without merging**